packages/tx/src/baseTransaction.ts

```diff
@@ -91,8 +91,20 @@ export abstract class BaseTransaction<TransactionObject> {
     const rB = toBuffer(r === '' ? '0x' : r)
     const sB = toBuffer(s === '' ? '0x' : s)
 
-    this.nonce = new BN(toBuffer(nonce === '' ? '0x' : nonce))
-    this.gasLimit = new BN(toBuffer(gasLimit === '' ? '0x' : gasLimit))
+    const bufferedNonce = toBuffer(nonce === '' ? '0x' : nonce)
+    if (bufferedNonce.length > 8) {
+      // Implements EIP-2681 which limits nonce to 2^64-1
+      // https://eips.ethereum.org/EIPS/eip-2681
+      throw new Error('nonce cannot be more than 2^64-1')
+    }
+    this.nonce = new BN(bufferedNonce)
+
+    const bufferedGasLimit = toBuffer(nonce === '' ? '0x' : gasLimit)
+    if (bufferedGasLimit.length > 8) {
+      throw new Error('gasLimit can not be more than 2^64-1')
```

Comment:
is this specified somewhere?

Reply:
Here's [the test](https://github.com/ethereum/tests/blob/develop/TransactionTests/ttGasLimit/TransactionWithGasLimitOverflow64.json) that fails if you don't put this change in.

Reply:
```suggestion
      throw new Error('gasLimit cannot be more than 2^64-1')
```
to match the above `nonceB` cannot